devices/voicecontroldevice.cpp

```diff
@@ -0,0 +1,47 @@
+#include "voicecontroldevice.h"
+#include <iostream>
+
+namespace smartHome {
+    VoiceControlDevice::VoiceControlDevice(const std::string& deviceId, bool status, const std::vector<Sensor>& sensors)
+        : Device("Voice Control", deviceId, status, sensors) {}
+
+    void VoiceControlDevice::powerOn() {
+        //exit methode if the device is already functioning
+        if(getStatus()){
+            return;
+        }
+        std::string keyword = "Activate";
```

Comment:
suggestion: make it at least const
suggestion 2: have both keywords defined as static const strings members of you class this way we avoid creating objects for each call (not a problem here but to get accustomed with the practice) when we can improve the performance a bit. Constexpr would have been preferable but unfortunately, we don't have constexpr strings in C++17 (20 has them but we're not yet there) and trying to make these values constexpr would give you more headaches.